store/sqlstore/channel_store_categories.go

```diff
@@ -14,22 +14,27 @@ import (
 	"github.com/mattermost/mattermost-server/v5/store"
 )
 
-func (s SqlChannelStore) CreateInitialSidebarCategories(userId, teamId string) error {
+func (s SqlChannelStore) CreateInitialSidebarCategories(userId, teamId string) (*model.OrderedSidebarCategories, error) {
 	transaction, err := s.GetMaster().Begin()
 	if err != nil {
-		return errors.Wrap(err, "CreateInitialSidebarCategories: begin_transaction")
+		return nil, errors.Wrap(err, "CreateInitialSidebarCategories: begin_transaction")
 	}
 	defer finalizeTransaction(transaction)
 
-	if err := s.createInitialSidebarCategoriesT(transaction, userId, teamId); err != nil {
-		return errors.Wrap(err, "CreateInitialSidebarCategories: createInitialSidebarCategoriesT")
+	if err = s.createInitialSidebarCategoriesT(transaction, userId, teamId); err != nil {
+		return nil, errors.Wrap(err, "CreateInitialSidebarCategories: createInitialSidebarCategoriesT")
+	}
+
+	oc, err := s.getSidebarCategoriesT(transaction, userId, teamId)
```

Comment:
Hmm .. so this is doing read after write again. We want to return the data from `createInitialSidebarCategoriesT` itself. We already know what we are inserting, so there is no need to read from the DB again.

Reply:
We are reading from master so I don't see issues as far as consistency goes. If we are concerned about the additional query I guess we can just transform the data directly in `createInitialSidebarCategoriesT()`.

Reply:
Correct. That's what I was trying to say. Sorry for any confusion. For any set of queries, that write and then read the same data back again, we should return the written value from the write method itself. There is no need for an extra read query.

Somewhat similar to this case here: https://community-daily.mattermost.com/core/pl/nphtmkowcjd8ic76tbqtapx6nc. Except that the create method doesn't even return anything here.

Reply:
I can try to do that. My only worry is that we might not have enough information in the create method to return all the data we require in which case we would need to run additional queries anyway. There's also the risk that the db state differs from what we expect so again the create wouldn't return a true representation of it.

Reply:
Discussed offline. It's not exactly a read-after-write problem. The existing approach looks great.